### ISSUE TITLE
400% Cpu usage by node due to meteor

### ISSUE THREAD

**Issue by MartiniHenry1988** [meteor/meteor#10445]
I have a meteor project, which is constantly going down because of extremely higher CPU usage.

As I start my project, my CPU usage jumps to 400+ percentage for node.

Currently using **METEOR@1.8.0.2**
OS: **Ubuntu 16.04**

I literally have no clue whats going on and my live product is being effected big time.

**Comment by StorytellerCZ** [comment]
Well, it is kind of hard to say what might be wrong with such a little info. Let's start with listing of packages (both npm and atmosphere) that your project uses. Do you use APM or some other analytics tool that shows what is going on before or during the CPU spike? What is your app doing that might cause a high CPU usage?

**Comment by MartiniHenry1988** [comment]
Here is my package.json:

```
{
  "name": "socially",
  "private": true,
  "scripts": {
    "start": "meteor run",
    "test": "meteor test --driver-package practicalmeteor:mocha",
    "test:ci": "meteor test --once --driver-package dispatch:mocha-phantomjs"
  },
  "dependencies": {
    "@babel/runtime": "^7.2.0",
    "apn": "2.2.0",
    "babel-runtime": "^6.23.0",
    "bcrypt": "3.0.3",
    "esendex": "0.1.8",
    "gcm": "1.0.1",
    "node-gcm": "0.14.4",
    "nodemailer": "2.0.0",
    "nodemailer-smtp-transport": "2.5.0",
    "qr-image": "3.1.0",
    "stripe": "^5.2.0"
  },
  "devDependencies": {
    "babel-plugin-angularjs-annotate": "^0.8.0"
  }
}
```

and here is meteor package file:

```
# Meteor packages used by this project, one per line.
# Check this file (and the other files in this directory) into your repository.
#
# 'meteor add' and 'meteor remove' will edit this file for you,
# but you can also edit it by hand.

meteor-base            # Packages every Meteor app needs to have
#mobile-experience@1.0.5       # Packages for a great mobile UX
mongo                   # The database Meteor supports right now
session                 # Client-side reactive dictionary for your app
jquery@1.11.10                # Helpful client-side library
#tracker                 	  # Meteor's client-side reactive programming library

es5-shim@4.8.0               # ECMAScript 5 compatibility for older browsers.

#email@1.2.3
angular
#twbs:bootstrap
fortawesome:fontawesome
angularui:angular-ui-router
accounts-password@1.5.1
#accounts-facebook@1.2.1
#service-configuration@1.0.11
aldeed:collection2
#meteortoys:allthings
fourseven:scss
jonatan:angular-google-places-autocomplete
http@1.4.1
rajit:bootstrap3-datepicker
#skys215:bootstrap-material-datetimepicker
#drewy:datetimepicker
#yuukan:streamy
#meteorhacks:npm
#npm-container
matthieuh:angular-google-maps-native
sathriawan:angular-timer
cfs:filesystem
cfs:standard-packages
cfs:graphicsmagick
jparker:crypto-md5
meteorhacks:ssr
#meteorhacks:aggregate
sakulstra:aggregate
#dschulz:jquery-qrcode
base64@1.0.11
jparker:crypto-sha256
mizzao:user-status
angularutils:pagination
#peerlibrary:xml2js
reywood:publish-composite
aldeed:moment-timezone
tmeasday:publish-counts
angular:angular-material
nimble:restivus
angularui:angular-ui-bootstrap
#gadicohen:phantomjs
#spiderable
planettraining:angular-translate
planettraining:angular-translate-loader-static-files
check@1.3.1
ostrio:logger@1.1.2
ostrio:loggerfile@1.1.2
kadira:dochead
#meteorhacks:kadira
#meteorhacks:kadira-profiler
#kadira:debug
swiper:swiper
#bigdsk:inputmask
aaronroberson:ng-mask
angular:angular-messages
interwebs:ui-sortable


#these packages specifically export to pdf 
nilsdannemann:pdfmake

#date-range-picker
fragaria:angular-daterangepicker

#copleykj:stripe-sync
standard-minifier-css@1.5.2
abernix:standard-minifier-js
shell-server@0.4.0
#facebook-config-ui@1.0.0
dynamic-import@0.5.0
dotansimha:accounts-ui-angular-moded
#practicalmeteor:mocha

mrt:googlemaps
#meteorhacks:fast-render
#appcache
underscore@1.0.10

#montiapm:agent
#montiapm:profiler
```

**Comment by veered** [comment]
How much load is your db under? How much memory is your server using?

**Comment by MartiniHenry1988** [comment]
Mongodb also spikes between 100 and 200

**Comment by MartiniHenry1988** [comment]
Here are the logs of MontiAPM:
Currently my server is showing 140% cpu usage, but MontiAPM is showing less.

![image](https://user-images.githubusercontent.com/14938690/52464897-72102a80-2b9e-11e9-839c-7abb9aced4bd.png)

![image](https://user-images.githubusercontent.com/14938690/52464911-82280a00-2b9e-11e9-93b5-51612d248620.png)

![image](https://user-images.githubusercontent.com/14938690/52465138-55c0bd80-2b9f-11e9-90ac-10a83fa06f86.png)

**Comment by sebakerckhof** [comment]
Well, obviously from the last 2 screenshots there's something on the client side flooding your servers with login requests. Either knowingly or because of a bug in your frontend.

**Comment by MartiniHenry1988** [comment]
@sebakerckhof But how can I debug this ? This is happening by a user which is using our android application. And Android application is using ddp: 

I never updated this library from past 3 years.

https://github.com/delight-im/Android-DDP

**Comment by rj-david** [comment]
We encountered an issue recently wherein if an app is connected and we deploy a new version of the app, the app kept on refreshing (the result was a ddos-like behavior flooding our servers) after the new version was up (we use a blue-green deployment wherein the load balancer will switch the connections to the new version when up and then bring down the old instances).

We still do not know why this is happening.

What we ended up doing was to have a settings publication and during deployment, our deployment script will automatically trigger an api call that will update the value of the settings publication. This change will force all connected apps to disconnect. Then the script will switch the balancer to the new instances. The apps will automatically re-connect after one minute. So far seems like it solved this problem of apps trying to refresh over and over from our tests. We will be testing more broadly tomorrow
